src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py

```diff
@@ -2782,6 +2782,26 @@ def test_get_enable_aad(self):
         with self.assertRaises(RequiredArgumentMissingError):
             ctx_3.get_enable_aad()
 
+        # custom value
+        ctx_4 = AKSContext(
+            self.cmd,
+            {
+                "enable_aad": True,
+            },
+            self.models,
+            decorator_mode=DecoratorMode.UPDATE,
+        )
+        aad_profile_4 = self.models.ManagedClusterAADProfile(
+            managed=True,
+        )
+        mc_4 = self.models.ManagedCluster(
+            location="test_location", aad_profile=aad_profile_4
+        )
+        ctx_4.attach_mc(mc_4)
+        # fail on managed aad already enabled
+        with self.assertRaises(InvalidArgumentValueError):
+            ctx_4.get_enable_aad()
+
```

Comment:
Do we need to test the context?

```
AKSContext(
            self.cmd,
            {
                "enable_aad": True,
                "aad_client_app_id": "test_aad_client_app_id",
            },
            self.models,
            decorator_mode=DecoratorMode.CREATE,
        )
```

Reply:
Yeah, we have a similar test case (the second test case in this function `test_get_enable_aad`) for the mutual exclusion between `enable_aad` and `aad_client_app_id/aad_server_app_id/aad_server_app_secret`.  Maybe you can expand a little bit above :)

Reply:
The newly added test case is for `aks update`, existing ones for `aks create`.